spring-cloud-contract-tools/spring-cloud-contract-spec-pact/src/main/groovy/org/springframework/cloud/contract/verifier/spec/pact/PactContractConverter.groovy

```diff
@@ -52,284 +37,32 @@ class PactContractConverter implements ContractConverter<Pact> {
 		}
-		return new RequestResponsePact(provider, consumer, interactions)
-	}
 
-	protected void assertInputContract(parsedJson) {
-		boolean hasExecutionProp = false
-		MapConverter.transformValues(parsedJson, {
-			if (it instanceof ExecutionProperty) {
-				hasExecutionProp = true
-			}
-			return it
-		})
-		if (hasExecutionProp) {
-			throw new UnsupportedOperationException("We can't convert a contract that has execution property")
-		}
+		throw new UnsupportedOperationException("We currently don't support pact contracts of type" + pact.class.simpleName)
 	}
 
-	protected Map<String, String> headers(Headers headers, Closure closure) {
-		return headers.entries.collectEntries {
-			String name = it.name
-			String value = closure(it)
-			return [(name) : value]
-		}
-	}
+	@Override
+	Collection<Pact> convertTo(Collection<Contract> contracts) {
+		List<Pact> pactContracts = new ArrayList<>()
 
-	protected Map<String, Map<String, Object>> matchingRules(BodyMatchers bodyMatchers) {
-		return bodyMatchers.jsonPathMatchers().collectEntries {
-			MatchingType matchingType = it.matchingType()
-			String key = "\$.body${it.path().startsWith('$') ? it.path().substring(1) : it.path()}"
-			Object value = it.value()
-			Integer minTypeOccurrence = it.minTypeOccurrence()
-			Integer maxTypeOccurrence = it.maxTypeOccurrence()
-			Map<String, Object> matchingRule = [:]
-			switch (matchingType) {
-				case MatchingType.EQUALITY:
-					matchingRule << [(MATCH_KEY) : MatchingType.EQUALITY.toString().toLowerCase() as Object]
-					break
-				case MatchingType.TYPE:
-					Map<String, Object> map = [(MATCH_KEY) : MatchingType.TYPE.toString().toLowerCase() as Object]
-					if (minTypeOccurrence) map.put(MIN_KEY, minTypeOccurrence)
-					if (maxTypeOccurrence) map.put(MAX_KEY, maxTypeOccurrence)
-					matchingRule << map
-					break
-				case MatchingType.DATE:
-				case MatchingType.TIME:
-				case MatchingType.TIMESTAMP:
-				case MatchingType.REGEX:
-					matchingRule << [
-							(MATCH_KEY) : MatchingType.REGEX.toString().toLowerCase() as Object,
-							(REGEX_KEY) : value
-					]
-					break
+		contracts.collect({ Contract contract ->
+			if (contract.request) {
+				pactContracts.add(requestResponsePactCreator.createFromContract(contract))
 			}
-			return [(key) : matchingRule]
-		}
-	}
 
-	protected String url(Contract dsl) {
-		if (dsl.request.urlPath) {
-			return dsl.request.urlPath.serverValue.toString()
-		} else if (dsl.request.url) {
-			return dsl.request.url.serverValue.toString()
-		}
-		throw new IllegalStateException("No url provided")
-	}
+			if (contract.input) {
+				pactContracts.add(messagePactCreator.createFromContract(contract))
+			}
+		})
 
-	protected QueryParameters queryParams(Contract dsl) {
-		if (dsl.request.urlPath) {
-			return dsl.request.urlPath.queryParameters
-		} else if (dsl.request.url) {
-			return dsl.request.url.queryParameters
-		}
-		throw new IllegalStateException("No url provided")
+		pactContracts
```

Comment:
missing `return` (I wasn't adding this comment everywhere since it doesn't make much sense to repeat myself)